ssl/t1_lib.c

```diff
@@ -1283,8 +1630,9 @@ static const SIGALG_LOOKUP *tls1_get_legacy_sigalg(const SSL_CONNECTION *s,
             size_t i;
 
             /* Work out index corresponding to ciphersuite */
-            for (i = 0; i < SSL_PKEY_NUM; i++) {
-                const SSL_CERT_LOOKUP *clu = ssl_cert_lookup_by_idx(i);
+            for (i = 0; i < s->ssl_pkey_num; i++) {
+                const SSL_CERT_LOOKUP *clu = ssl_cert_lookup_by_idx(i,
+				                     SSL_CONNECTION_GET_CTX(s));
```

Comment:
This looks nicer, I'd say:

```suggestion
                const SSL_CERT_LOOKUP *clu
                    = ssl_cert_lookup_by_idx(i, SSL_CONNECTION_GET_CTX(s));
```

Reply:
so changed in https://github.com/openssl/openssl/pull/19312/commits/1f20a63fbcb8f733651546e40328ae2ab2090537